src/main/java/com/google/api/codegen/discovery/transformer/java/JavaSampleMethodToViewTransformer.java

```diff
@@ -107,8 +109,11 @@ public SampleBodyView createSampleBody(SampleTransformerContext context) {
             "method is page streaming, but the page streaming resource field is null.");
       }
       sampleBodyView.resourceGetterName(sampleNamer.getResourceGetterName(fieldInfo.name()));
-      sampleBodyView.resourceTypeName(
-          sampleTypeTable.getAndSaveNickNameForElementType(fieldInfo.type()));
+      String resourceTypeName = sampleTypeTable.getAndSaveNickNameForElementType(fieldInfo.type());
+      String resourceVarName =
+          sampleNamer.getResourceVarName(fieldInfo.type().isMessage() ? resourceTypeName : "");
+      sampleBodyView.resourceVarName(symbolTable.getNewSymbol(resourceVarName));
+      sampleBodyView.resourceTypeName(resourceTypeName);
```

Comment:
nit: since this only depends on `resourceTypeName`, move to just after the `String resourceTypeName = ...` line.